src/OpenCV/Photo.hs

```diff
@@ -205,26 +213,33 @@ fastNlMeansDenoisingColoredMultiImg = exceptError $ do
 
 fastNlMeansDenoisingColoredMulti
    :: Double -- ^ Parameter regulating filter strength for luminance component.
-             -- Bigger h value perfectly removes noise but also removes image details,
-             -- smaller h value preserves details but also preserves some noise
-   -> Double -- ^ The same as h but for color components. For most images value equals 10 will be enough
-             -- to remove colored noise and do not distort colors
-   -> Int32  -- ^ templateWindowSize Size in pixels of the template patch that is used to compute weights.
-             -- Should be odd. Recommended value 7 pixels
-   -> Int32  -- ^ searchWindowSize. Size in pixels of the window that is used to compute weighted average
-             -- for given pixel. Should be odd. Affect performance linearly:
-             -- greater searchWindowsSize - greater denoising time.
-             -- Recommended value 21 pixels
-   -> V.Vector (Mat ('S [h, w]) ('S 3) ('S Word8)) -- ^ Vector of odd number of input 8-bit 3-channel images.
-   -> CvExcept (Mat ('S [h, w]) ('S 3) ('S Word8)) -- ^ Output image same size and type as input.
-
-fastNlMeansDenoisingColoredMulti h hColor templateWindowSize searchWindowSize srcVec = unsafeWrapException $ do
+             -- Bigger h value perfectly removes noise but also removes image
+             -- details, smaller h value preserves details but also preserves
+             -- some noise
+   -> Double -- ^ The same as h but for color components. For most images value
+             -- equals 10 will be enough to remove colored noise and do not
+             -- distort colors
+   -> Int32  -- ^ templateWindowSize Size in pixels of the template patch that
+             -- is used to compute weights. Should be odd.
+             -- Recommended value 7 pixels
+   -> Int32  -- ^ searchWindowSize. Size in pixels of the window that is used to
+             -- compute weighted average for given pixel. Should be odd.
+             -- Affect performance linearly: greater searchWindowsSize -
+             -- greater denoising time. Recommended value 21 pixels
+   -> V.Vector (Mat ('S [h, w]) ('S 3) ('S Word8))
```

Comment:
Instead of dropping the last image when the length of the vector is not odd, why not throw an error?

I think it is better to give an explicit error than to silently ignore a wrong input. You can even add a constructor to the `CvException` type in `OpenCV.Internal.Exception`. Maybe something like `PreconditionError String` or `ParameterError String`.